concepts/50 React Components/70 TypeScript Support/10 Kinds of Types/05 Event Objects.md

```diff
@@ -0,0 +1,74 @@
+Each component has event types that define arguments for handling events. Naming convention is the following: the argument type of the `contentReady` event handler is `ContentReadyEvent`.
+
+Event types are based on common parts:
+
+    export type AppointmentClickEvent = Cancelable & NativeEventInfo<dxScheduler, KeyboardEvent | MouseEvent | PointerEvent> & TargetedAppointmentInfo & {
+        readonly appointmentElement: DxElement;
+    };
+
+### Common Event Parts
+
+Events are union types built from common and specific parts:
+
+    export type AppointmentTooltipShowingEvent = Cancelable & EventInfo<dxScheduler> & {
+        readonly targetElement: DxElement;
+        readonly appointments: AppointmentTooltipShowingAppointmentInfo[];
+    };
```

Comment:
Without example of howto read read understand types it's a bit pointless

> AppointmentTooltipShowing event object is common synthetic event object, because it’s based on EventInfo, that can be canceled ad has two specific fields: targetElement and appointments

>CellContextMenu event object is common native event object, cause it’s based on NativeEventInfo. It can’t be canceled, native event can be one of the following: MouseEvent | PointerEvent | TouchEvent. There are two specific fields as well: cellData and cellElement.